client/src/components/Common/MapEditor/index.js

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from "react";
+import Geocode from "react-geocode";
+import styled from "styled-components";
+
+import { spacing32 } from "../../../resources/style-constants";
+import InputFormField from "../InputFormField";
+import Map from "../Map";
+
+const Container = styled.div`
+  width: 100%;
+  height: 30vw;
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: flex-start;
+`;
+
+const MapContainer = styled.div`
+  height: 100%;
+  width: calc(60% - ${spacing32});
+  margin-right: ${spacing32};
+`;
+
+const OptionsContainer = styled.div`
+  height: 100%;
+  width: 40%;
+`;
+
+/**
+ * Displays a map with an edit field beside it. Use this
+ * to provide a GUI to update a maps geolocation (lat, lng).
+ * 
+ * IMPORTANT: For this api to work, you'll have to enable
+ * the geocoding API for your API key.
+ * 
+ * @label
+ * The text which appears above the input field.
+ * 
+ * @onLocationChanged
+ * A callback invoked whenever the location for the map changes.
+ * 
+ * ```
+ * <MapEditor 
+ *    label={'My Label'} 
+ *    onLocationChanged={(location) => {
+ *      console.log(`latitude:  ${location.lat}`);
+ *      console.log(`longitude: ${location.lng}`)
+ *    }}
+ * />
+ * ```
+ * 
+ */
+const MapEditor = ({
+  label,
+  onLocationChanged,
+}) => {
+  const [address, setAddress] = useState("");
+  const [location, setLocation] = useState(undefined);
+  const [validationState, setValidationState] = useState({
+    address: {
+      error: true,
+      message: ""
+    },
+  });
+
+  const getLocationFromAddress = async function (value) {
```

Comment:
Are we going to need this method for the other map on the event view page as well? If so, we may want to create a MapHelper.js and place this method in that file and export it. Then on this page and the event view we would just import the file from MapHelper.